pkg/log/zap.go

```diff
@@ -0,0 +1,51 @@
+package log
+
+import (
+	"os"
+
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	ctrlruntimezap "sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type Logger struct {
+	*zap.SugaredLogger
+}
+
+func New(debug bool) *Logger {
+	return &Logger{newZappLogger(debug).Sugar()}
+}
+
+func (k *Logger) WithError(err error) log.Logger {
```

Comment:
How would this be used?

Reply:
This is used in the virtual kubelet thread, https://github.com/rancher/k3k/pull/133/files#diff-e53078009a4f5b2cec981fb4d89b9895034c9e083e9342f44ae5ac8c6f755effR100, the virtual kubelet logger needs this interface to be implemented